examples/components/BANK_MARKETING_VERTICAL/traininsilo/host.py

```diff
@@ -0,0 +1,551 @@
+"""Script for mock components."""
+import argparse
+import logging
+import sys
+import copy
+import os
+from aml_comm import AMLCommRedis
+
+import mlflow
+import torch
+import pandas as pd
+import numpy as np
+from torch import nn
+from torchmetrics.functional import precision_recall, accuracy
+from torch.optim import SGD
+from torch.utils.data.dataloader import DataLoader
+from typing import List
+import models as models
+import datasets as datasets
+
+
+class RunningMetrics:
+    def __init__(self, metrics: List[str], prefix: str = None) -> None:
+        self._allowed_metrics = copy.deepcopy(metrics)
+        self._running_step_metrics = {name: 0 for name in metrics}
+        self._running_global_metrics = {name: 0 for name in metrics}
+        self._batch_count_step = 0
+        self._batch_count_global = 0
+        self._prefix = "" if prefix is None else prefix
+
+    def add_metric(self, name: str, value: float):
+        """Add measurement to the set
+
+        Args:
+            name: name of the metrics
+            value: value of the metrics
+        """
+        if name not in self._allowed_metrics:
+            raise ValueError(f"Metric with name '{name}' not in logged metrics")
+        self._running_step_metrics[name] += value
+        self._running_global_metrics[name] += value
+
+    def step(self):
+        """Increases number of measurements taken. Must be called after every batch"""
+        self._batch_count_step += 1
+        self._batch_count_global += 1
+
+    def reset_step(self):
+        """Soft reset of the counter. Only reset counter for subset of batches."""
+        self._batch_count_step = 0
+        self._running_step_metrics = {name: 0 for name in self._running_step_metrics}
+
+    def reset_global(self):
+        """Reset all counters and steps"""
+        self.reset_step()
+        self._batch_count_global = 0
+        self._running_global_metrics = {
+            name: 0 for name in self._running_global_metrics
+        }
+
+    def get_step(self):
+        """Provide average value for every metric since last `reset_step` call"""
+        return {
+            f"{self._prefix}_step_{name}": value / self._batch_count_step
```

Comment:
in ccfraud vertical you're using prefix_name_running, and here prefix_step_name, any reason?